packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx

```diff
@@ -112,4 +120,87 @@ const getDocumentStatus = (
   return docStatus;
 };
 
-export { LocalePickerAction };
+/* -------------------------------------------------------------------------------------------------
+ * DeleteLocaleAction
+ * -----------------------------------------------------------------------------------------------*/
+
+const DeleteLocaleAction: DocumentActionComponent = ({ document, id, model, collectionType }) => {
+  const { formatMessage } = useIntl();
+  const navigate = useNavigate();
+  const toggleNotification = useNotification();
+  const { delete: deleteAction } = useDocumentActions();
+  const { hasI18n, canDelete } = useI18n();
+
+  if (!hasI18n) {
+    return null;
+  }
+
+  return {
+    disabled: (document?.locale && !canDelete.includes(document.locale)) || !document,
+    position: ['header', 'table-row'],
+    label: formatMessage({
+      id: getTranslation('actions.delete.label'),
+      defaultMessage: 'Delete locale',
+    }),
+    icon: <StyledTrash />,
+    variant: 'danger',
+    dialog: {
+      type: 'dialog',
+      title: formatMessage({
+        id: getTranslation('actions.delete.dialog.title'),
+        defaultMessage: 'Confirmation',
+      }),
+      content: (
+        <Flex direction="column" gap={2}>
+          <Icon as={ExclamationMarkCircle} width="24px" height="24px" color="danger600" />
+          <Typography as="p" variant="omega" textAlign="center">
+            {formatMessage({
+              id: getTranslation('actions.delete.dialog.body'),
+              defaultMessage: 'Are you sure?',
+            })}
+          </Typography>
+        </Flex>
+      ),
+      onConfirm: async () => {
+        if (!id || !document?.locale) {
+          console.error(
+            "You're trying to delete a document without an id or locale, this is likely a bug with Strapi. Please open an issue."
```

Comment:
I've seen this kind of condition in a few PRs. This situation shouldn't be possible right? Are these here just in case the app ends up in this state somehow?

Reply:
exactly that, because we can't really assert through typescript when these conditions happen because a document action technically renders in the create locale too because we want to show them (which imo is right to do) we just make sure we do something to also signify whats happened to users. We also know we have a big issue if it does happen lol